### DISCUSSION TITLE
GitHub Actions: Execute a script file without checkout

### DISCUSSION THREAD

**Discussion by cwdbox** [community/community#26359] [Actions]
<p>Just starting with GitHub Actions. Is there a way of bundling script files to be referenced and executed as part of a Step without checking out the repo?</p>
<pre><code class="lang-yaml">steps:
  - name: Run Script
    shell: bash
    run: ./my-script.sh
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>Not as script files, no, you’ll only have access to that after checkout. What’s the issue with checking out the repository?</p>
<p>Two possible alternatives, but not exactly what you’re asking for:</p>
<ol>
<li>Include the script directly in the <code>run</code> block, if it’s not too big.</li>
</ol>
<pre><code class="lang-yaml">steps:
  - name: Run Script
    shell: bash
    run: |
      ls this/
      echo "that"
</code></pre>
<ol start="2">
<li>Set up and use an <a href="https://docs.github.com/en/free-pro-team@latest/actions/creating-actions">Action</a> that runs your script (probably a Docker or Composite action). Not that running an action implicitly checks out the action repository, but in that case you can keep the script in the action repository.</li>
</ol>

**Comment by cwdbox** [comment]
<p>Thanks. I was exploring GitHub Actions capabilities and I have a workflow that builds, tests and pushes a docker image to a registry. Then I have another workflow that deploys a container on a remote host based on a tag (don’t want to deploy it on every change to a branch). Since the image is already in the registry I have no need to checkout the repo to produce the image. In case of big repo (for example) just seemed a bit wasteful to checkout. The reasoning was also to keep the workflow declaration small and concise by pushing the logic onto script files. I understand there are alternatives, like a dedicated repo or a custom Action.</p>
